integrationTests/chainSimulator/bridge/esdtSafe_test.go

```diff
@@ -0,0 +1,518 @@
+			Identifier: tokensMapper[token.Identifier],
+			Nonce:      token.Nonce,
+			Amount:     token.Amount,
+			Type:       token.Type,
+		}
+		if isSftOrMeta(receivedToken.Type) {
+			chainSim.RequireAccountHasToken(t, cs, getTokenIdentifier(receivedToken), esdtSafeAddr, big.NewInt(1))
+		} else {
+			chainSim.RequireAccountHasToken(t, cs, getTokenIdentifier(receivedToken), esdtSafeAddr, big.NewInt(0))
+		}
+
+		waitIfCrossShardProcessing(cs, esdtSafeAddrShard, receiverShardId)
+		chainSim.RequireAccountHasToken(t, cs, getTokenIdentifier(receivedToken), receiver.Bech32, receivedToken.Amount)
+
+		// --------------------------------------------
+		// deposit back 1 token, expect token to be burned and supply burned is 1
+		depositAmount := big.NewInt(1)
+		depositToken := chainSim.ArgsDepositToken{
+			Identifier: receivedToken.Identifier,
+			Nonce:      receivedToken.Nonce,
+			Amount:     depositAmount,
+			Type:       receivedToken.Type,
+		}
+		txResult = deposit(t, cs, receiver.Bytes, &receiverNonce, bridgeData.ESDTSafeAddress, []chainSim.ArgsDepositToken{depositToken}, wallet.Bytes)
+		chainSim.RequireSuccessfulTransaction(t, txResult)
+		chainSim.RequireAccountHasToken(t, cs, getTokenIdentifier(receivedToken), receiver.Bech32, big.NewInt(0).Sub(receivedToken.Amount, big.NewInt(1)))
+
+		waitIfCrossShardProcessing(cs, esdtSafeAddrShard, receiverShardId)
+
+		expectedAmount := big.NewInt(0)
+		if isSftOrMeta(receivedToken.Type) {
+			// for (dynamic) SFT/MetaESDT, the contract should always have 1 esdt
+			expectedAmount = big.NewInt(1)
+		}
+		chainSim.RequireAccountHasToken(t, cs, getTokenIdentifier(depositToken), esdtSafeAddr, expectedAmount)
+
+		_ = cs.GenerateBlocks(1)
+		tokenSupply, err := cs.GetNodeHandler(esdtSafeAddrShard).GetFacadeHandler().GetTokenSupply(getTokenIdentifier(depositToken))
+		require.Nil(t, err)
+		require.NotNil(t, tokenSupply)
+		require.Equal(t, depositAmount.String(), tokenSupply.Burned)
+
+		nextShardId(&receiverShardId)
+	}
+
+	// generate hello contracts in each shard
+	// hello contract has one endpoint "hello" which receives a number as argument
+	// if number is 0 then will throw error, otherwise will do nothing
+	receiverContracts := generateHelloInAllShards(t, cs)
+
+	// transfer sovereign chain -> main chain with transfer data
+	// execute 2nd time the same operations
+	// for (dynamic) NFT, expect creation with nonce+1 the second time because that's the next nonce
+	// for (dynamic) SFT/MetaESDT, the contract should just add quantity for existing nonce
+	for _, token := range tokens {
+		// get contract from next shard
+		receiver := receiverContracts[receiverShardId]
+
+		// execute operations received from sovereign chain
+		// expecting the token to be minted in esdt-safe contract with the same properties and transferred to receiver address
+		trnsData := &transferData{
+			GasLimit: uint64(10000000),
+			Function: []byte("hello"),
+			Args:     [][]byte{{0x01}},
```

Comment:
Why do we need this contract with 0 numbers if we always call it with non zero value?

Reply:
I am not sure what you mean.
It's just a bridge transaction with transfer data to a SC on endpoint `hello` with one number as argument.
- if number > 0, then the call is success --> this is the current test
- if number == 0, then the call should fail --> this is under investigation because is not working properly, will create a test in another PR

Reply:
That's what I was concerned about, because you never call it with 0 here and didn't see the point.
But since you said you'll use it in another PRs, it's fine